apps/potterhat_metrics/lib/potterhat_metrics/statix_reporter.ex

```diff
@@ -0,0 +1,259 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule PotterhatMetrics.StatixReporter do
+  @moduledoc """
+  Reports events to Statix backend.
+  """
+  use Statix, runtime_config: true
+
+  @behaviour PotterhatUtils.TelemetrySubscriber
+
+  @supported_events [
+    [:periodic_metrics, :active_nodes, :collected],
+    [:periodic_metrics, :configured_nodes, :collected],
+    [:active_nodes, :registered],
+    [:active_nodes, :deregistered],
+    [:rpc, :request, :start],
+    [:rpc, :request, :stop],
+    [:rpc, :request, :success],
+    [:rpc, :request, :failed],
+    [:rpc, :request, :failed_over],
+    [:event_listener, :new_head, :subscribe_success],
+    [:event_listener, :new_head, :subscribe_failed],
+    [:event_listener, :new_head, :head_received],
+    [:event_listener, :log, :subscribe_success],
+    [:event_listener, :log, :subscribe_failed],
+    [:event_listener, :log, :log_received],
+    [:event_listener, :new_pending_transaction, :subscribe_success],
+    [:event_listener, :new_pending_transaction, :subscribe_failed],
+    [:event_listener, :new_pending_transaction, :transaction_received],
+    [:event_listener, :sync_status, :subscribe_success],
+    [:event_listener, :sync_status, :subscribe_failed],
+    [:event_listener, :sync_status, :sync_started],
+    [:event_listener, :sync_status, :sync_stopped]
+  ]
+
+  @impl true
+  def init do
+    __MODULE__.connect()
+  end
+
+  @impl true
+  def supported_events, do: @supported_events
+
+  #
+  # Periodic metrics
+  #
+
+  @impl true
+  def handle_event([:periodic_metrics, :active_nodes, :collected], measures, meta, _config) do
+    _ = gauge("potterhat.active_nodes.total_active", measures.total, opts(meta))
+  end
+
+  @impl true
+  def handle_event([:periodic_metrics, :configured_nodes, :collected], measures, meta, _config) do
+    _ = gauge("potterhat.nodes.total_configured", measures.total, opts(meta))
+  end
+
+  #
+  # Active nodes
+  #
+
+  @impl true
+  def handle_event([:active_nodes, :registered], _measures, meta, _config) do
+    _ = increment("potterhat.active_nodes.num_registered", 1, opts(meta))
+  end
+
+  @impl true
+  def handle_event([:active_nodes, :deregistered], _measures, meta, _config) do
+    _ = increment("potterhat.active_nodes.num_deregistered", 1, opts(meta))
+  end
+
+  #
+  # RPC requests
+  #
+
+  @impl true
+  def handle_event([:rpc, :request, :start], _measures, meta, _config) do
+    eth_method = meta.conn.assigns[:eth_method]
```

Comment:
I'd prefer if we just pass `eth_method` directly rather than passing `conn`. Note that Erlang VM actually copy the whole `conn` across processes when doing this, which can be quite expensive to GC.

Reply:
Unfortunately this is also a [`Plug.Telemetry`](https://github.com/elixir-plug/plug/blob/v1.8.2/lib/plug/telemetry.ex#L74-L83) thing.